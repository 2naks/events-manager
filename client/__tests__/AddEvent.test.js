import ConnectedAddEvent, { AddEvent } from '../components/AddEvent.jsx';

describe('AddEvent Component', () => {
  const initialState = {
    eventReducer: {
      status: "Error",
      message: "",
    }
  };


  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const props = {
    dispatch: () => {},
    show: () => {}
  };

  describe('AddEvent form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<AddEvent {...props}/>);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render for connected component', () => {
      const wrapper = shallow(<Provider store={store}><ConnectedAddEvent /></Provider>);
      expect(wrapper.length).toEqual(1);
    });

    it('it should render redirect on  success', () => {
      const wrapper = shallow(<AddEvent {...props} status="Success" />);
      expect(wrapper.length).toEqual(1);
    });
  });

  describe('AddEvent events', () => {
    it('it should submit form', () => {
      const wrapper = mount(<Provider store={store}><ConnectedAddEvent /></Provider>);
      const button = wrapper.find('.btn');
      button.simulate('click', { preventDefault: () => {} });
    });
  });
});
