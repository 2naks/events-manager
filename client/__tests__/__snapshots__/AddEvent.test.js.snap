// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEvent Component AddEvent form  should render self it should render for dumb component 1`] = `
<div>
  <NavBar
    page="AddEvent"
  />
  <div
    className="container add-event "
  >
    <div
      className="row"
    >
      <div
        className="container"
      >
        <form>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor=""
            />
            <div
              className="col-sm-9"
            />
          </div>
          <p
            className="text-center h5"
          >
            <b>
              ADD EVENT
            </b>
          </p>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Title
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Title"
                type="text"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="Description"
            >
              Description
            </label>
            <div
              className="col-sm-9"
            >
              <textarea
                className="form-control"
                name="description"
                onChange={[Function]}
                placeholder="Description"
                type="text"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="venue"
            >
              Venue
            </label>
            <div
              className="col-sm-9"
            >
              <select
                className="form-control"
                name="centerId"
                onChange={[Function]}
              >
                <option />
              </select>
              <small
                className="form-text text-muted"
                id="fileHelp"
              >
                <a
                  href="./centers"
                  target="_blank"
                >
                  View Centers
                </a>
              </small>
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Start Date
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="startDate"
                onChange={[Function]}
                type="date"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              End Date
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="endDate"
                onChange={[Function]}
                type="date"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="title"
            >
              Time
            </label>
            <div
              className="col-sm-9"
            >
              <input
                className="form-control"
                name="time"
                onChange={[Function]}
                type="time"
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="exampleInputFile"
            >
              Image
            </label>
            <div
              className="col-sm-9"
            >
              <input
                aria-describedby="fileHelp"
                className="form-control"
                name="image"
                onChange={[Function]}
                type="file"
              />
              <small
                className="form-text text-muted"
                id="fileHelp"
              >
                ( Optional )
              </small>
            </div>
          </div>
          <div
            className="form-group row"
          >
            <div
              className="offset-sm-3 col-sm-9"
            >
              <button
                className="btn btn-sm btn-success"
                onClick={[Function]}
                type="submit"
              >
                <i
                  className="fa fa-plus fa-lg"
                />
                 Add Event
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
